src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/caseupdater/CcdBundleCloningService.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.em.orchestrator.service.caseupdater;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.em.orchestrator.service.ccdcallbackhandler.CcdCallbackDto;
+import uk.gov.hmcts.reform.em.orchestrator.service.dto.CcdBundleDTO;
+import uk.gov.hmcts.reform.em.orchestrator.service.dto.CcdValue;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class CcdBundleCloningService implements CcdCaseUpdater {
+
+    private final ObjectMapper objectMapper;
+    private final JavaType type;
+
+    public CcdBundleCloningService(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+        type = objectMapper.getTypeFactory().constructParametricType(CcdValue.class, CcdBundleDTO.class);
+    }
+
+    @Override
+    public boolean handles(CcdCallbackDto ccdCallbackDto) {
+        return false;
+    }
+
+    @Override
+    public JsonNode updateCase(CcdCallbackDto ccdCallbackDto) {
+
+        Optional<ArrayNode> maybeBundles = ccdCallbackDto.findCaseProperty(ArrayNode.class);
+        List<JsonNode> updatedBundlesList = new ArrayList<>();
+
+        if (maybeBundles.isPresent()) {
+            for (int i = 0; i < maybeBundles.get().size(); i++) {
```

Comment:
use a `for (JsonNode originalJson : bundles)` unless you need the i index for something